query-service/build.gradle.kts

```diff
@@ -27,7 +27,12 @@ dependencies {
 
   integrationTestImplementation("org.apache.kafka:kafka-clients:5.5.1-ccs")
   integrationTestImplementation("org.apache.kafka:kafka-streams:5.5.1-ccs")
-  integrationTestImplementation("org.apache.avro:avro:1.11.0")
+  integrationTestImplementation("org.apache.avro:avro:1.11.1")
+  constraints {
+    implementation("com.fasterxml.jackson.core:jackson-databind:2.13.4.2") {
```

Comment:
same question - avoid constraints except as last resort, who is bringing in this jackson version?

Reply:
It is coming from avro.
```
Issues with no direct upgrade or patch:
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038424] in com.fasterxml.jackson.core:jackson-databind@2.12.7
    introduced by org.apache.avro:avro@1.11.1 > com.fasterxml.jackson.core:jackson-databind@2.12.7 and 2 other path(s)
  This issue was fixed in versions: 2.13.4
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038426] in com.fasterxml.jackson.core:jackson-databind@2.12.7
    introduced by org.apache.avro:avro@1.11.1 > com.fasterxml.jackson.core:jackson-databind@2.12.7 and 2 other path(s)
  This issue was fixed in versions: 2.12.7.1, 2.13.4.2
```

Reply:
all the apache projects' dependency handling really is just problematic. So here, this is an integration test dep and you're modifying the runtime classpath to fix - we can omit that from the scan instead. The runtime should already be fixed, too since we constrained this elsewhere.

Reply:
This seems to compile time dependency - https://mvnrepository.com/artifact/org.apache.avro/avro/1.11.1

Do you think we can go ahead with this constraint?

Reply:
So avro itself here appears to be an integration test dependency - if so, we're adding an actual runtime constraint for a test dependency.

Reply:
Missed that part. Removed it.